### ISSUE TITLE
New Rule: ("tsconfig 'target' and eslint env do not match. Update target and env to same ECMAScript version ex. es2021")

### ISSUE THREAD

**Issue by BboyAkers** [eslint/eslint#16488]
### Rule details

When using es2021 methods in typescript, eslint doesn't automatically detect this and gives an error of type 'any'

### Related ECMAScript feature

no-unsafe-call

### What type of rule is this?

Warns about a potential problem

### Example code

```js
.tsconfig
{
  "compilerOptions": {
    "target":"es2021"
}

.eslintrc
{
  "env": {
    "browser": true,
    "node": false
    //no "es2021":true added here
  },
}


const colors = {$palette-blue-10:"#d7eaf8",$palette-blue-20:"#9ccbee",$palette-blue-30:"#75b6e7"};

const formatColorPalette = (palette: Record<string, string>) => {
	return Object.keys(palette).reduce((acc, key) => {
		const oldKey = key;
		key = key.replace('$palette-', '');
                // error message that should be shown when hovering over the .replaceAll()
                // tsconfig 'target' and eslint env do not match. Update target and env to same ECMAScript version ex. es2021
		key = key.replaceAll('-', '');
		acc[key] = palette[oldKey];
		return acc;
	}, {} as Record<string, string>);
};

console.log(formatColorPalette(colors));
```

### Why should this rule be in the core instead of a plugin?

Many developers use typescript and eslint. This would definitely make for a better debugging experience if we receive errors stemming from the .tsconfig and the .eslintrc ECMAScript version not matching. When using .replaceAll() I received a "no-unsafe-call" error from eslint even I was using .replaceAll() on a variable that was a string.

### Participation

- [X] I am willing to submit a pull request to implement this rule.

### Additional comments

I'm not sure where to start but I'd love to get some pointers, dive in, and figure out how to make this PR happen! 🙂

**Comment by mdjermanovic** [comment]
Hi @BboyAkers, thanks for the issue!

This is rather a question for typescript-eslint. Core ESLint doesn't access `.tsconfig` files, and [`no-unsafe-call`](https://typescript-eslint.io/rules/no-unsafe-call/) is a rule from `@typescript-eslint/eslint-plugin`.

You can open an issue in this repository:

https://github.com/typescript-eslint/typescript-eslint

**Comment by BboyAkers** [comment]
Will do! Thanks a ton for leading me in the right direction @mdjermanovic 🙂
